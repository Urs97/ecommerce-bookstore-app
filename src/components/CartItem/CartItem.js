import React, { useRef } from 'react';
import './CartItem.scss';
import test_photo from '../../assets/images/testimonial-portrait-3.jpg';

const CartItem = ({ header }) => {
    const quantityInputRef = useRef(null);

    const handleUpArrow = event => {
        event.preventDefault();
        quantityInputRef.stepUp(1);
    }

    const handleDownArrow = event => {
        event.preventDefault();
        quantityInputRef.stepDown(1);
    }

    if(header) return (
        <section className="cart-item cart-item-header">
            <span></span>
            <h3 className="title">Image</h3>
            <h3 className="title">Product</h3>
            <h3 className="title">Price</h3>
            <h3 className="title">Quantity</h3>
            <h3 className="title">Subtotal</h3>
        </section>
    )

    // Add button focus ring functionality

    return (
        <section className="cart-item">
            <button className="remove-item">X</button>
            <span className="item-img"><img src={test_photo} alt="product"/></span>
            <span className="item-title">
                <h4>Nineteen Eighty-Four</h4>
                <p>George Orwell</p>
            </span>
            <span className="item-price">$21.00</span>
            <form className="item-quantity">
                <label htmlFor="quantity" className="screen-reader-only">item-title quantity</label>
                <input ref={quantityInputRef} type="number" id="quantity" name="quantity" 
                    step="1" min="1" max="99" defaultValue="1" />
                <section className="item-quantity-btns">
                    <button onClick={ handleUpArrow } className="plus">▲</button>
                    <button onClick={ handleDownArrow } className="minus">▼</button>
                </section>
            </form>
            <span className="item-subtotal">$21.00</span>
        </section>
    );
};

export default CartItem;