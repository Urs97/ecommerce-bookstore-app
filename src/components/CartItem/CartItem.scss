@import '../../styles/shared.scss';

.cart-item {
    display: grid;
    grid-template-columns: 0.5fr minmax(max-content, 1fr) 3fr repeat(3, minmax(max-content, 1fr));
    grid-template-rows: 10em;

    .remove-item {
        cursor: pointer;
        outline: none;
        border: none;
        font-size: 1.5em;
        font-weight: 600;
        color: red;
        margin-left: 1em;
    }

    .item-img {
        img {
            width: 5em;
        }
    }
    
    .item-title {
        display: flex;
        flex-direction: column;
    }

    .item-quantity {
        display: inline-flex;

        input {
            padding: 0.5em 0.5em;
            background-color: rgba(29, 187, 144, 0.3);
            width: 2.75em;
            height: 3em;
            outline: none;
            border: none;
            border-top-left-radius: $border-radius;
            border-bottom-left-radius: $border-radius;
            overflow: hidden;
            white-space: nowrap;
            text-align: end;
        }

        .item-quantity-btns {
            display: flex;
            flex-direction: column;

            button {
                padding: 0.2em 0.1em;
                color: black;
                outline:none;
                -webkit-appearance: none;
                background-color: rgba(29, 187, 144, 0.3);
                border: none;
                align-items: center;
                justify-content: center;
                width: 1.5em;
                height: 1.5em;
                cursor: pointer;
                margin: 0;
                position: relative;
              }
              
            & .up-arrow {
                padding-right: 1em;
                border-top-right-radius: $border-radius;
            }

            & .down-arrow {
                padding-right: 1em;
                border-bottom-right-radius: $border-radius;
            }
        }
    }

// .number-input,
// .number-input * {
//   box-sizing: border-box;
// }

    .item-subtotal {
        font-weight: 600;
    }
}

.cart-item > * {
    display: flex;
    align-items: center;
    justify-content: center;
    background-color: #FFFFFF;
}

.cart-item-header {
    grid-template-rows: 5em;
    border: none;
    border-radius: $border-radius $border-radius 0 0;
    background-color: $green;
}

.cart-item-header > * {
    background-color: $green;
}

.cart-item-header :first-child {
    border-top-left-radius: $border-radius;
}

.cart-item-header :last-child {
    border-top-right-radius: $border-radius;
}