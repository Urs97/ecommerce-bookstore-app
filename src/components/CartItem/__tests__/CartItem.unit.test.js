import React from 'react';
import {shallow} from 'enzyme';
import CartItem from '../CartItem';

describe('CartItem', () => {
    test('renders empty CartItem', () => {
        const wrapper = shallow(
            <CartItem />
        );
        expect(wrapper).toMatchSnapshot();
    });
});