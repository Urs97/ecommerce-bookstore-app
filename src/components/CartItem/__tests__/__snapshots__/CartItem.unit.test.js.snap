// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem renders empty CartItem 1`] = `
<section
  className="cart-item"
>
  <span
    className="remove-item-container"
  >
    <button
      aria-label="Remove item"
      className="remove-item"
      onClick={[Function]}
    >
      X
    </button>
  </span>
  <span
    className="item-img"
  >
    <img
      alt="Frankenstein book cover"
      src="https://covers.openlibrary.org/b/olid/OL6147070M-L.jpg"
    />
  </span>
  <span
    className="item-title"
  >
    <h4>
      Frankenstein
    </h4>
    <p>
      Marry Shelley
    </p>
  </span>
  <span
    className="item-price"
  >
    $
    94.99
  </span>
  <form
    className="item-quantity"
    onSubmit={[Function]}
  >
    <label
      className="screen-reader-only"
      htmlFor="quantity"
    >
      item-title quantity
    </label>
    <input
      id="quantity"
      max="99"
      min="1"
      name="quantity"
      onChange={[Function]}
      step="1"
      type="number"
      value={1}
    />
    <section
      className="item-quantity-btns"
    >
      <button
        aria-label="Add 1 item unit"
        className="up-arrow"
        onClick={[Function]}
      >
        ▲
      </button>
      <button
        aria-label="subtract 1 item unit"
        className="down-arrow"
        onClick={[Function]}
      >
        ▼
      </button>
    </section>
  </form>
  <span
    className="item-subtotal"
  >
    $
    94.99
  </span>
</section>
`;
